mlir/include/mlir/Dialect/Quartz/IR/QuartzOps.td

```diff
@@ -0,0 +1,378 @@
+// Unitary Operations
+//===----------------------------------------------------------------------===//
+
+def XOp : QuartzOp<"x", traits = [Quartz_UnitaryOpInterface]> {
+    let arguments = (ins QubitType:$qubit_in);
+    let assemblyFormat = "$qubit_in attr-dict";
+
+    let extraClassDeclaration = [{
+        size_t getNumQubits() { return getNumTargets() + getNumControls(); }
+        static size_t getNumTargets() { return 1; }
+        size_t getNumControls() { return getNumPosControls() + getNumNegControls(); }
+        static size_t getNumPosControls() { return 0; }
+        static size_t getNumNegControls() { return 0; }
+        Value getQubit(size_t i);
+        Value getTarget(size_t i);
+        static Value getPosControl(size_t i);
+        static Value getNegControl(size_t i);
+        static size_t getNumParams() { return 0; }
+        static ParameterDescriptor getParameter(size_t i);
+        static bool hasStaticUnitary() { return true; }
+        DenseElementsAttr tryGetStaticMatrix();
+        static StringRef getBaseSymbol() { return "x"; }
+    }];
+}
+
+def RXOp : QuartzOp<"rx", traits = [Quartz_UnitaryOpInterface]> {
+    let arguments = (ins QubitType:$qubit_in,
+                     OptionalAttr<F64Attr>:$theta_attr,
+                     Optional<F64>:$theta_operand);
+    let assemblyFormat = "`(` ($theta_attr^)? ($theta_operand^)? `)` $qubit_in attr-dict";
+
+    let extraClassDeclaration = [{
+        size_t getNumQubits() { return getNumTargets() + getNumControls(); }
+        static size_t getNumTargets() { return 1; }
+        size_t getNumControls() { return getNumPosControls() + getNumNegControls(); }
+        static size_t getNumPosControls() { return 0; }
+        static size_t getNumNegControls() { return 0; }
+        Value getQubit(size_t i);
+        Value getTarget(size_t i);
+        static Value getPosControl(size_t i);
+        static Value getNegControl(size_t i);
+        static size_t getNumParams() { return 1; }
+        ParameterDescriptor getParameter(size_t i);
+        bool hasStaticUnitary();
+        DenseElementsAttr tryGetStaticMatrix();
+        static StringRef getBaseSymbol() { return "rx"; }
+    }];
+
+    let builders = [
+        OpBuilder<(ins "Value":$qubit_in, "double":$theta_double), [{
+            auto theta_attr = FloatAttr::get(Float64Type::get($_builder.getContext()), theta_double);
+            build($_builder, $_state, qubit_in, theta_attr, nullptr);
+        }]>,
+        OpBuilder<(ins "Value":$qubit_in, "FloatAttr":$theta_attr), [{
+            build($_builder, $_state, qubit_in, theta_attr, nullptr);
+        }]>,
+        OpBuilder<(ins "Value":$qubit_in, "Value":$theta_operand), [{
+            build($_builder, $_state, qubit_in, nullptr, theta_operand);
+        }]>,
+    ];
+
+    let hasVerifier = 1;
+}
+
+def U2Op : QuartzOp<"u2", traits = [Quartz_UnitaryOpInterface, AttrSizedOperandSegments]> {
```

Comment:
What's the benefit of the `AttrSizedOperandSegments` trait here?
Is it necessary for something? Does it add some value?